account_payment_partner/models/account_move.py

```diff
@@ -90,13 +99,8 @@ def get_bank_id():
                         bank_id = get_bank_id()
             move.invoice_partner_bank_id = bank_id
 
-    # I think copying payment mode from invoice to refund by default
-    # is a good idea because the most common way of "paying" a refund is to
-    # deduct it on the payment of the next invoice (and OCA/bank-payment
-    # allows to have negative payment lines since March 2016)
     def _reverse_move_vals(self, default_values, cancel=True):
         move_vals = super()._reverse_move_vals(default_values, cancel=cancel)
-        move_vals["payment_mode_id"] = self.payment_mode_id.id
```

Comment:
I think that we should obtain the refund payment mode from the invoice payment mode.
If we do not assign anything, default refund partner payment mode will be set.

You can add a test in which the invoice has a different payment mode than the default customer payment mode to check what I say.

Reply:
There are 2 cases:

- Doing a refund from an invoice. In that case, indeed the payment mode from original invoice should be checked and use the refund payment mode for that one.
- Doing a manual refund or invoicing a sales order which results in a refund. In this case (which is the one I had in mind and the customer user case), the payment mode should be extracted from the partner, and the test should be done in the partner onchange.

Reply:
Of course, that's why I make the point here inside the _reverse_move_vals method, which is used in case 1.

Reply:
This one is still missing to fill the refund with the refund payment mode from the original invoice payment mode.